### DISCUSSION TITLE
Is there an easy way to connect to Google Cloud Shell using Teleport?

### DISCUSSION THREAD

**Discussion by 621625** [gravitational/teleport#8629] [Q&A]
Hello,

I'm BM. A very nice and polite guy.

Is there an easy way to connect to Google Cloud Shell using Teleport?

**Comment by stevenGravy** [comment]
We presently support AWS console (7.0) and will support AWS Cli for (8.0).  gcp console/shell may come in a future version.

**Comment by 621625** [comment]
Ok. If you remember, please ping me in this thread if you ever support Google Cloud Shell.

By the way, isn't the point of Teleport that it should make servers accessible everywhere? Isn't that the goal? Not just specific platforms I mean.

I have used Syncthing successfully on Google Cloud Shell.

By the way, how does Teleport work? Do you route through decentralized nodes?

-------- Original Message --------

**Reply by stevenGravy** [reply]
Teleport uses a agent or integrating to a openssh server to provide SSH access. The web or cli access for a particular cloud is thus not required for users to access a server.  Typically that's more for users administrating within the respective cloud.  Here is a detailed how it works guide. https://goteleport.com/teleport/how-it-works/
